payjoin/tests/integration.rs

```diff
@@ -45,13 +45,52 @@ mod integration {
             // **********************
             // Inside the Receiver:
             // this data would transit from one party to another over the network in production
-            let response = handle_pj_request(req, headers, receiver);
+            let valid_response = handle_pj_request(req, headers, receiver);
             // this response would be returned as http response to the sender
 
             // **********************
             // Inside the Sender:
+            //
+            // Validate sender can handle known errors correctly
+            let known_error = r#"{"errorCode":"version-unsupported", "message":"Debug Message"}"#;
+            let response_error =
+                ctx.clone().process_response(&mut known_error.as_bytes()).unwrap_err();
+            assert_eq!(
+                format!("{:#}", response_error),
+                "This version of payjoin is not supported."
+            );
+            assert_eq!(
+                format!("{:?}", response_error),
+                "Well known error: { \"errorCode\": \"version-unsupported\",\n                \"message\": \"Debug Message\" }"
+            );
+            match response_error {
+                payjoin::send::ResponseError::WellKnown(e) => {
+                    assert_eq!(e.clone().error_code(), "version-unsupported");
+                    assert_eq!(e.message(), "Debug Message");
+                }
+                _ => panic!("Unexpected error type"),
+            };
```

Comment:
I definitely prefer the match asserts to the assert_eq. Seems like that assert_eq could start failing while the response is still spec valid.